File: OJ/Leetcode/Algorithm/90. Subsets II.cpp
class Solution {
public:
    vector<vector<int>> subsetsWithDup(vector<int>& nums) {
        int n = nums.size();
        int N = (1<<n);
        set<vector<int> > s;
        sort(nums.begin(), nums.end());
        for(int i = 0; i < N; i++)
        {
            vector<int> tmp;
            for(int j = 0; j < n; j++)
            {
                if(i & (1<<j))
                    tmp.push_back(nums[j]);
            }
            s.insert(tmp);
        }
        vector<vector<int> > ans(s.begin(), s.end());
        return ans;
    }
};

File: OJ/Leetcode/Algorithm/90. Subsets II.py
class Solution(object):
    def subsetsWithDup(self, nums):
        """
        :type nums: List[int]
        :rtype: List[List[int]]
        """
        res = [[]]
        nums.sort()
        for i in range(len(nums)):
            if i == 0 or nums[i] != nums[i - 1]:
                l = len(res)
            for j in range(len(res) - l, len(res)):
                res.append(res[j] + [nums[i]])
        return res